packages/api/README.md

```diff
@@ -39,12 +54,126 @@ the ability to merge schemas, a place to store resolvers in services, and authen
 
 ## Contributing
 
-Explains how to contribute by addressing the following three points:
+`@redwoodjs/api` uses a few things you should be familiar with:
+
+- [Apollo Server](https://www.apollographql.com/docs/apollo-server/) (Lambda)
+- [GraphQL](https://graphql.org/learn/)
+
+Although this package depends, in the code-dependency sense, only on `@redwoodjs/internals`, it still hangs together with the others&mdash;notably, `@redwoodjs/web` and `@redwoodjs/dev-server`. So if you’re asking yourself “but when does my server run?” head over to `@redwoodjs/dev-server`.
+
+### Overview
+
+To give the `@redwoodjs/api` context, we'll be referring to the Redwood-App file [graphql.js](https://github.com/redwoodjs/create-redwood-app/blob/master/api/src/functions/graphql.js), where functions from `@redwoodjs/api` are imported and invoked.
+Remember, files in this directory (`api/src/functions`) are serverless functions. This one in particular is a GraphQL API, and right now it’s what most of `@redwoodjs/api` is for&mdash;setting up the GraphQL API Redwood Apps come with by default. It all happens in essentially four steps:
+
+1. Everything (i.e. sdl and services) is imported
+2. The services are wrapped into resolvers
+3. The sdl and resolvers are merged/stitched into a schema
+4. The ApolloServer is instantiated with said merged/stitched schema and context
+
+These four steps map quite nicely to these four functions, some of which are doing a little more under-the-hood than others:
+
+1. `importAll`
+2. `makeServices`
+3. `makeMergedSchema`
+4. `createGraphQLHandler`
+
+### `importAll`
+
+In Redwood Apps, the resolvers are mapped automatically. The schemas are in the `./api/src/graphql` directory and the resolvers are in the `./api/src/services` directory, and as long as you export the right things with the right names from the right files, they get merged together.
+
+All of that magic actually happens in your Redwood App, in `./api/src/functions/graphql.js`. And since everything’s separated, the first thing Redwood does is import it all. And `importAll` does just that.
+
+`importAll` is a [babel macro](https://babeljs.io/blog/2017/09/11/zero-config-with-babel-macros). This one in particular imports files that match a glob ([thanks to Kent C. Dodds](https://github.com/kentcdodds/import-all.macro)). The difference between Kent’s and ours is that we normalize the keys for the files that are found:
+
+```javascript
+const services = importAll('api', 'services')
+
+// services = { 'todo': { // services/todos.js exports } }
+```
```

Comment:
We're actually going to remove this in favour of the babel dir importer, but this is amazing. <3

Reply:
You actually wrote half of this so that's probably the amazing part :sweat_smile:. I'll keep an eye out for that change modify this section when the time comes.